src/com/can/CMakeLists.txt

```diff
@@ -9,44 +9,51 @@
 # SPDX-License-Identifier: EPL-2.0
 # 
 # Contributors:
-#     Malte Grave - initial API and implementation and/or initial documentation
+#     Malte Grave, Alexander Trojnin - initial API and implementation and/or initial documentation
 # *******************************************************************************/
 # *******************************************************************************/
 #############################################################################
 # CAN Com Layer (currently only usable on Linux)
 #############################################################################
 
-function(CHECK_IF_KERNEL_MODULE_IS_LOADED)
-  execute_process(
-    COMMAND lsmod
-    COMMAND grep -wE "^(can|can_raw|vcan|can_dev)"
-    COMMAND awk "{print $1}"
-    COMMAND tr "\n" " "
-    OUTPUT_VARIABLE CAN_MODULES_FOUND
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
-  message(STATUS "Found CAN modules: ${CAN_MODULES_FOUND}")
+forte_add_network_layer(CAN OFF "can" CCANComLayer CANLayer "Enable the CAN Commuication Layer")
+set(FORTE_COM_CAN_MODULE_CHECK ON CACHE BOOL "Checking for the Presence of CAN Modules.")
```

Comment:
Last thing that I noticed, could you wrap the set(...) into an if(...) so is only gets checked when the layer is active? This wouldn't interfere with your use case and with this we ensure that this option is only valid when the CAN layer is selected. 

```cmake
if(FORTE_COM_CAN)
    set(FORTE_COM_CAN_MODULE_CHECK ON CACHE BOOL "Checking for the Presence of CAN Modules.")
endif()
```